docs/flow/config-language/components.md

```diff
@@ -6,3 +6,157 @@ weight: 300
 ---
 
 # Components
+Components are the defining feature of Grafana Agent Flow. They are autonomous
+pieces of business logic that perform a single task (like retrieving secrets or
+collecting Prometheus metrics) and are wired together to form programmable
+pipelines of telemetry data.
+
+Under the hood, components are orchestrated via the [_component
+controller_]({{< relref "../concepts/component_controller.md" >}}), who is
+responsible for scheduling them, reporting their health and debug status, and
+continuously re-evaluating their inputs and outputs.
+
+Each top-level River _block_ will instantiate a new component. A component
+consists of its name (which describes what the component is responsible for) as
+well as an optional user-specified label.
+
+You can see a list of all available components [here]({{< relref "../components/_index.md" >}}).
+Each component features a complete reference page, so getting a component to
+work for you should be as easy as reading its documentation and copy/pasting
+from an example.
+
+## Arguments and Exports
+Most user interactions with components will center around two basic concepts;
+_Arguments_ and _Exports_.
+
+* _Arguments_ work much like function arguments in your favorite programming
+ language. They can be any number of attributes or nested unlabeled blocks and
+are used to configure the component's behavior. Most components will feature
+both required and optional arguments. Any arguments that are not provided will
+take on their default values.
```

Comment:
Not all of our users will know how to write code; can we change analogies here? I think we can think of them as settings which modify the behavior of the component.